test/evergreen/tsan_warnings_analysis.py

```diff
@@ -25,55 +25,93 @@
-current_dir = os.getcwd()
-existing_tsan_options = os.environ.get("TSAN_OPTIONS")
-os.environ["TSAN_OPTIONS"] = f"{existing_tsan_options}:log_path={current_dir}/tsan_logs:exitcode=0"
+    :param file_path: Path to the file to analyze.
+    :param git_rev: Git revision (default: HEAD).
+    :return: List of tuples (line_number, line_content, last_modified_timestamp).
+    """
+    # Run `git blame` to retrieve metadata for found line number
+    git_command = ["git", "blame", "--line-porcelain", "HEAD", f"-L {line_number},{line_number}", "--", file_path]
+    result = subprocess.run(git_command, capture_output=True, text=True, check=True)
 
-# Start with examples suite.
-test_tasks = [
-    "ctest -j 8 --test-dir examples/c"
-]
+    timestamp = None
+    output_lines = result.stdout.splitlines()
+    for line in output_lines:
+        # Extract the UNIX timestamp
+        if line.startswith("author-time"):
+            timestamp = int(line.split()[1])
+    return timestamp
 
-# Run the tasks in list.
-print("Tasks to be executed")
-print(test_tasks)
-for task in test_tasks:
-    try:
-        # Tasks are allowed to run with maximum of three hours.
-        subprocess.run(task.split(), check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=60 * 60 * 3)
-    except subprocess.CalledProcessError as exception:
-        print(f'Command {exception.cmd} failed with error {exception.returncode}')
-    except subprocess.TimeoutExpired as exception:
-        print(f'Command {exception.cmd} timed out')
-        sys.exit(1)
+def get_tsan_warnings():
+    """
+    Get the TSAN warnings from the log files.
+    :return: Set of unique TSAN warnings.
+    """
+    tsan_warnings_set = set()
+    current_dir = os.getcwd()
 
-# Loop through WT root directory and search for tsan logs.
-tsan_warnings_set = set()
-for tsan_log in os.listdir(current_dir):
-    # Check if the file starts with "tsan"
-    if tsan_log.startswith("tsan"):
-        with open(tsan_log, "r") as file:
-            for line in file:
-                if (not line.startswith("SUMMARY:")):
-                    continue
-                # Strip away the unnecessary information
-                pattern_to_remove = r"/data/mci/.*/wiredtiger/"
-                cleaned_text = re.sub(pattern_to_remove, "", line).strip()
+    # Loop through WT root directory and search for tsan logs.
+    for root, _, files in os.walk(current_dir):
+        for file_name in files:
+            # Check if the file starts with "tsan"
+            if file_name.startswith("tsan_logs"):
+                file_path = os.path.join(root, file_name)  # Get the full path to the file
+                with open(file_path, "r") as file:
+                    for line in file:
+                        if (not line.startswith("SUMMARY:")):
+                            continue
+                        # Strip away the unnecessary information
+                        pattern_to_remove = r"/data/mci/.*/wiredtiger/"
```

Comment:
I think it would be great to have an ability to run this script locally on a workstation. Could we avoid relying on `/data/mci/` evergreen path?